rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
I find that confusing and inconsistent. If histogram_quantile is to be implemented as an aggregator (with the additional gotcha that it always swallows the `le` label), it should work as every other aggregator:

`sum(v) by (instance)` yields the sum for each instance.

`sum(v)` yields the sum of everything.

`histogram_quantile(.9,  rate(histo[5m])) by (instance)` yields the 0.9-quantile for each instance.

`histogram_quantile(.9,  rate(histo[5m]))` yields the 0.9-quantile for everything.